File: service/shared/src/main/java/com/simp/service/shared/data/clients/AuthClient.java
package com.simp.service.shared.data.clients;

import com.simp.service.shared.data.clients.feign.AuthClientFeign;
import com.simp.service.shared.server.payload.dto.AuthorizationDto;
import com.simp.service.shared.service.scheme.AuthControllerScheme;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;
import reactor.core.publisher.Mono;

@Component
@RequiredArgsConstructor
public class AuthClient implements AuthControllerScheme {
    private final AuthClientFeign feign;

    @Override
    public Mono<AuthorizationDto> getAuthData(String token) {
        return feign.getAuthData(token);
    }
}

File: service/shared/src/main/kotlin/com/simp/service/shared/data/clients/AuthClient.kt
package com.simp.service.shared.data.clients

import com.simp.service.shared.contants.Services
import com.simp.service.shared.domain.payload.GreetingsResponse
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam

@FeignClient(name = "authClient", url = "\${${Services.AUTH}}")
@ConditionalOnProperty(Services.AUTH)
interface AuthClient {
    @GetMapping("/greetings")
    fun greetings(@RequestParam("name") name: String): GreetingsResponse
}
